src/components/common/form/NumberInputField.tsx

```diff
@@ -0,0 +1,98 @@
+import { InputAdornment } from '@mui/material'
+import FormTextField from './FormTextField'
+
+import { isInteger, useField } from 'formik'
+import { useTranslation } from 'next-i18next'
+import { moneyPublic } from 'common/util/money'
+
+type Props = {
+  name?: string
+  limit?: number
+}
+
+export default function NumberInputField({
+  name = 'otherAmount',
+  limit = Number.MAX_SAFE_INTEGER,
+}: Props) {
+  const { t, i18n } = useTranslation('one-time-donation')
+  const [, meta, { setValue, setError }] = useField(name)
+  const decimalSeparator = (1.1).toLocaleString(i18n.lang).charAt(1)
+
+  return (
+    <FormTextField
+      name={name}
+      type="number"
+      value={meta.value === 1 ? '' : meta.value}
+      label={t('first-step.amount')}
+      lang={i18n.language}
+      onKeyDown={(e) => {
+        if (meta.error && e.key !== 'Backspace' && e.key !== 'Delete' && !isInteger(meta.value)) {
+          e.preventDefault()
+          return
+        }
+        if (decimalSeparator !== e.key && (e.key === '.' || e.key === ',')) {
+          e.preventDefault()
+          return
+        }
+
+        if (
+          (e.key.charCodeAt(0) >= 48 && e.key.charCodeAt(0) <= 57) ||
+          (isInteger(meta.value) && e.key === decimalSeparator) ||
+          (e.ctrlKey && e.key === 'v') ||
+          (e.ctrlKey && e.key === 'c') ||
+          e.key === 'Backspace' ||
+          e.key === 'Delete' ||
+          (e.ctrlKey && e.key === 'a') ||
```

Comment:
I've tested after the last changes and noticed that on Mac the Copy/Paste/SelectAll combinations are not working. The ctrlKey on Mac is e.metaKey - could you include in the logic too.

Reply:
Done, but I am unable to test it.

Reply:
works now!